hive-agent/src/main/java/org/apache/ranger/authorization/hive/authorizer/RangerHiveAuthorizer.java

```diff
@@ -3127,12 +3127,28 @@ private RangerHiveAccessRequest buildRequestForAlterTableSetOwnerFromCommandStri
 		return request;
 	}
 
+	void setOwnerUser(RangerHiveResource resource, HivePrivilegeObject hiveObj) {
+		if (hiveObj != null) {
+			if (StringUtils.isNotBlank(hiveObj.getOwnerName())) {
+				resource.setOwnerUser(hiveObj.getOwnerName());
+				return;
+			}
+		}
+		setOwnerUserInternal(resource, hiveObj, getMetaStoreClient());
+	}
+
 	static void setOwnerUser(RangerHiveResource resource, HivePrivilegeObject hiveObj, IMetaStoreClient metaStoreClient) {
 		if (hiveObj != null) {
 			if (StringUtils.isNotBlank(hiveObj.getOwnerName())) {
 				resource.setOwnerUser(hiveObj.getOwnerName());
 				return;
 			}
+		}
+		setOwnerUserInternal(resource, hiveObj, metaStoreClient);
+	}
+
+	private static void setOwnerUserInternal(RangerHiveResource resource, HivePrivilegeObject hiveObj, IMetaStoreClient metaStoreClient) {
+		if (hiveObj != null) {
```

Comment:
This `if` nesting seems to be redundant because check `hiveObj != null` already occured before calling `setOwnerUserInternal`.

Reply:
My bad :)